### ISSUE TITLE
Handling of `version` by backuity/clist

### ISSUE THREAD

**Issue by thpani** [informalsystems/apalache#1624]
We are currently parsing CLI arguments using [backuity/clist](https://github.com/backuity/clist).
We recently had some problems with how the `version` command is handled, see #1497, #1505, and #1611:

* We cannot parse access CLI arguments, in particular `--debug`, when Apalache is run with these commands (#1497, #1505). However, we use `version` as a no-op command, e.g. in integration tests.
* The built-in commands `version` and `help` are not printed in the usage / `--help` output (#1611).

This is an architectural shortcoming of `clist`, which does not internally represent the built-ins `version` and `help` as `Command`s.
In particular, for accessing flags, it returns an `Option[Command]` from parsing, and `None` on the built-ins.

We should file an upstream bug or consider another CLI parser.

**Comment by konnov** [comment]
> We should file an upstream bug or consider another CLI parser.

Yep, `clist` has annoying behavior. Perhaps, there is a newer better CLI parser for scala.

**Comment by shonfeder** [comment]
Definitely room for improvement here, i think the key question is around prioritization.

**Comment by shonfeder** [comment]
The most actively developed and attractive (IMO) CLI parsing libs I've found so far are

- https://github.com/bkirwi/decline
- https://github.com/alexarchambault/case-app

Scallop also looks pretty good, and it has more stars, but it hasn't been updated quite as recently (could just be cause it's mature and stable tho):

- https://github.com/scallop/scallop

Finally, `scopt` is the most heavily stared library I've seen, but the API doesn't look wonderful to me:

- https://github.com/scopt/scopt

**Comment by shonfeder** [comment]
`case-app` could be desirable as it could enable automating much of the bookkeeping required to integrate with the configuration system.

**Comment by konnov** [comment]
This should be fairly easy to fix with LLMs now. I would also prefer a Java library that has a history of long-term maintenance.

**Comment by thpani** [comment]
The well-maintained JVM/Java options seem to be:

- https://github.com/remkop/picocli
- https://commons.apache.org/proper/commons-cli/changes.html

**Comment by konnov** [comment]
`picocli` looks especially interesting, as they encourage the projects to include their source code, which aims at having small footprint.
